api/samples/CMakeLists.txt

```diff
@@ -172,6 +172,20 @@ function (add_sample_client name source_file_list extension_list)
     "${sample_list};${name}")                                     # NON-PUBLIC
   set(srcs ${srcs} ${source_file_list} PARENT_SCOPE)              # NON-PUBLIC
   set(tgts ${tgts} ${name} PARENT_SCOPE)                          # NON-PUBLIC
+
+  # Some samples actually have more tests than just simple sanity tests. These
+  # tests require that results are shown. In order to avoid short-circuiting
+  # the SHOW_RESULTS option, we build an additional sample (which shows results)
+  # for testing purposes.
+  if (BUILD_TESTS)
+    add_library(${name}_test SHARED ${source_file_list})
+    configure_DynamoRIO_client(${name}_test)
+    foreach (ext ${extension_list})
+      use_DynamoRIO_extension(${name}_test ${ext})
+    endforeach (ext)
+    target_compile_definitions(${name}_test PRIVATE -DSHOW_RESULTS)
```

Comment:
But today we want the _test version to *not* have SHOW_RESULTS for our sanity tests of running an app and having zero output (long-term yes we want output for _test, but console and not popup for Windows, and functional tests for each sample, but we're not there yet).

How about this:
+ The _test target does *not* have SHOW_RESULTS: i.e., it's silent
+ Today's sanity tests of these samples use the _test target
+ Your new opcode_count functional test uses the regular (non _test) target and is only on UNIX?
+ Add comments (file issue?) about removing the SHOW_RESULTS CMake var (we only need a cpp define), converting the others to use functional tests, and changing _test to replace the Windows popup with console printing

Reply:
Wait so your version still has BUILD_TESTS turning off SHOW_RESULTS so it's reversed from mine?

Reply:
Yes, it is reversed.

Reply:
I would vote for reversing what this PR has: it is more work (remove the disabling or whatnot and tweak the sanity tests) but it would be nice to have the regular sample show results in a build dir w/ tests (I've been annoyed by that before).

Reply:
It has to be reversed. If we set the main exported sample to always show results when building tests, we will ignore the user's SHOW_RESULTS option.

Reply:
Or if you want to put that under an issue which also covers removing SHOW_RESULTS and longer-term stuff that seems ok.

Reply:
> we will ignore the user's SHOW_RESULTS option.

I want to get rid of that option -- but sure, a separate issue + PR.

I think this needs a comment (ideally w/ issue ref) that just enabling SHOW_RESULTS is not enough for testing: we have to change the Windows popup to be a console print.

Reply:
Okay, I filed https://github.com/DynamoRIO/dynamorio/issues/4392